index.html

```diff
@@ -671,16 +909,60 @@
 	let output = `Generated by,${author}, \nDate,${curDate}, \n
 	Pitch Deviations are from ET\n
 	Width Deviations are from Just\n\n
-	cPitchETDev,ePitchETDev,5M WidthJIDev,3M WidthJIDev,Comment, \n`;
+	row&col,cPitchETDev,ePitchETDev,5M WidthJIDev,3M WidthJIDev\n`;
+
+	// slotData.forEach( slot => {
+	// 	output += `${slot.cFromET}, ${slot.eFromET}, ${slot.fifthDeviation}, ${slot.thirdDeviation}, ${slot.comment}\n`;
+	// })
 
-	slotData.forEach( slot => {
-		output += `${slot.cFromET}, ${slot.eFromET}, ${slot.fifthDeviation}, ${slot.thirdDeviation}, ${slot.comment}\n`;
-	})
+	// changes : I removed the 'comment' column because I am using it as a way to specify columns. I added an 'id' column that contains
+	// information about rows and columns of slots. 
+	TuningMap.forEach(function(value, key, map){
+			output += `${key},${value.cFromET}, ${value.eFromET}, ${value.fifthDeviation}, ${value.thirdDeviation}\n`;
+		});
 
 	let blob = new Blob([output], {type: "text/csv;charset=utf-8"});
 	saveAs(blob, author+"-"+curDate+".csv");
 })
 
+// imports data from a csv file
+let importData = () => {
+	// 	clearTableAndMap();
+
+	// 	Koi, use this template to import from a file
+
+	// 	for(all rows){
+
+	// 		var data = row.data;
+	// 		var id = row.id;
+	// 		TuningMap.set(id, data);
+	// 		var data = getRowAndColumnOfButton(id);
+	// 		var row = preferenceTable.rows[data[0]];
+	// 		var cell = row.cells[data[1]];
+	// 		cell.innerHTML = "<button id = " + "\"" + id +  "\"" + "> " + "</button>";
+
+	// 	}
+ }
+
+
+// removes all buttons from the slot interface and clears the map we are using to store data.
+let clearTableAndMap = () => {
+	TuningMap.forEach(function(value, key, map){
+			if(key != null){
+				var data = getRowAndColumnOfButton(key);
+				var row = preferenceTable.rows[data[0]];
+				var cell = row.cells[data[1]];
+				cell.innerHTML = '';
+			}
+		});
```

Comment:
HTMLTableElement (the type of preferenceTable), has the method `deleteRow`.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableElement/deleteRow

This would be preferable over manipulating the HTML directly.